### PR TITLE
BOOYAH-18 - orm basics

### PR DESCRIPTION
**Pull request by marceloribeiro**
Solves #18

### CODE REVIEW

File: src/config/routes.json [hunk 1]
```diff
@@ -1,4 +1,10 @@
 [
+  { "delete": "/users/{id}", "to": "users#destroy", "format": "json" },
```
Comment (klaus-trabalhando) [inline src/config/routes.json:2]: Maybe you could change the format, avoiding to use {}, I think should avoid some characters to be able to use regexp routes in the future.

File: src/db/adapters/postgresql_adapter.py [hunk 2]
```diff
@@ -44,13 +47,71 @@ def execute(self, query):
     def fetch(self, query):
         self.connect()
         cursor = self.connection.cursor()
-        cursor.execute(query)
+        logger.debug("DB: ", query)
```
Comment (klaus-trabalhando) [inline src/db/adapters/postgresql_adapter.py]: Do not need to use spaces when using more than one arg in logger, the default delimiter is space.

File: src/db/adapters/postgresql_adapter.py [hunk 3]
```diff
@@ -44,13 +47,71 @@ def execute(self, query):
...
+        try:
+            cursor.execute(query)
+        except Exception as e:
+            logger.fatal("DB FETCH: ", e)
```
Comment (klaus-trabalhando) [inline src/db/adapters/postgresql_adapter.py]: Remove unnecessary space "DB FETCH:"

File: src/db/adapters/postgresql_adapter.py [hunk 4]
```diff
@@ -44,13 +47,71 @@ def execute(self, query):
...
+        cursor.execute(query)
         columns = [item for row in cursor.fetchall() for item in row]
-        return columns
\ No newline at end of file
+        return columns
+
+    def last_insert_id(self, table_name):
+        query = f"SELECT id FROM {table_name} ORDER BY id DESC LIMIT 1"
+        records = self.fetch(query)
+        if len(records) == 0:
```
Comment (klaus-trabalhando) [inline src/db/adapters/postgresql_adapter.py]: Maybe you can use:
`if not records:`

# bad
if len(a) > 0:  # evaluates to True
  print("the list is not empty!")
  
"the list is not empty!"

# good
if a: # evaluates to True
  print("the list is not empty!")

File: src/db/adapters/postgresql_adapter.py [hunk 5]
```diff
@@ -44,13 +47,71 @@ def execute(self, query):
...
+            return None
+        return records[0][0]
+
+    def insert(self, table_name, attributes):
+        if attributes.get('created_at') == None:
```
Comment (klaus-trabalhando) [inline src/db/adapters/postgresql_adapter.py]: maybe you can use is None instead of ==

File: src/db/adapters/postgresql_adapter.py [hunk 6]
```diff
@@ -44,13 +47,71 @@ def execute(self, query):
...
+            attributes['created_at'] = datetime.now()
+        if attributes.get('updated_at') == None:
+            attributes['updated_at'] = datetime.now()
+
+        columns = ', '.join(attributes.keys())
+        values = ', '.join([f"'{value}'" for value in attributes.values()])
```
Comment (klaus-trabalhando) [inline src/db/adapters/postgresql_adapter.py]: I think we should have a method to convert attribute values to real sql values by checking datatype, something like:

def formatted_value(self, value):
    if value is Datetime:
      return "'" + value.strftime(...) + "'"
    if value is Numbers:
    ...
    if value is File:
       return Base64.encode(value.read)....
    else
       return "'" + str(value).replace("'", "\\'") + "'"

File: src/db/adapters/postgresql_adapter.py (hunk 2 above)
Comment (marceloribeiro) [reply src/db/adapters/postgresql_adapter.py]: @klaus-trabalhando this is solved now

File: src/db/adapters/postgresql_adapter.py (hunk 4 above)
Comment (marceloribeiro) [reply src/db/adapters/postgresql_adapter.py]: @klaus-trabalhando this is solved now

File: src/db/adapters/postgresql_adapter.py (hunk 6 above)
Comment (marceloribeiro) [reply src/db/adapters/postgresql_adapter.py]: @klaus-trabalhando this is solved now

File: src/db/adapters/postgresql_adapter.py (hunk 5 above)
Comment (marceloribeiro) [reply src/db/adapters/postgresql_adapter.py]: @klaus-trabalhando this is solved now

File: src/db/adapters/postgresql_adapter.py (hunk 3 above)
Comment (marceloribeiro) [reply src/db/adapters/postgresql_adapter.py]: @klaus-trabalhando this is solved now

**Review by thauan-andre-clear — APPROVED** [review]

**Pull request merged**